MdeModulePkg/Include/Guid/DxeMemoryProtectionSettings.h

```diff
@@ -0,0 +1,543 @@
+  //                                      section becomes non-executable.
+  //  .ProtectImageFromFv               : If set, images from firmware volumes will be protected by DxeCore
+  //                                      if they are aligned. The code section becomes read-only, and the data
+  //                                      section becomes non-executable.
+  //  .RaiseErrorIfProtectionFails      : If set, the image protection logic will return an error if protection fails.
+  //  .BlockImagesWithoutNxFlag         : If set, images of subsystem type EFI_APPLICATION
+  //                                      which don't utilize the /NXCOMPAT DLL flag will not be loaded.
+  //
+  // Note: If a bit is cleared, an image data section could be still non-executable if
+  // NxProtectionPolicy is enabled for EfiLoaderData, EfiBootServicesData or EfiRuntimeServicesData.
+  DXE_IMAGE_PROTECTION_POLICY    ImageProtectionPolicy;
+
+  // Indicates which type allocation need guard page.
+  //
+  // If bit is set, a head guard page and a tail guard page will be added just
+  // before and after corresponding type of pages which the allocated pool occupies,
+  // if there's enough free memory for all of them. The pool allocation for the
+  // type related to cleared bits keeps the same as ususal.
+  //
+  // This bitfield is only valid if UefiPoolGuard and/or MmPoolGuard are set in HeapGuardPolicy.
+  DXE_HEAP_GUARD_MEMORY_TYPES    HeapGuardPoolType;
+
+  // Indicates which type allocation need guard page.
+  //
+  // If a bit is set, a head guard page and a tail guard page will be added just
+  // before and after corresponding type of pages allocated if there's enough
+  // free pages for all of them. The page allocation for the type related to
+  // cleared bits keeps the same as ususal.
+  //
+  // This bitfield is only valid if UefiPageGuard is set in HeapGuardPolicy.
+  DXE_HEAP_GUARD_MEMORY_TYPES    HeapGuardPageType;
+
+  // DXE no execute memory protection policy.
+  //
+  // If a bit is set, memory regions of the associated type will be mapped
+  // non-executable. If a bit is cleared, nothing will be done to associated type of memory.
+  DXE_HEAP_GUARD_MEMORY_TYPES    NxProtectionPolicy;
+
+  // Indicates if stack cookie protection will be enabled
+  //
+  // A stack cookie check failure will trigger an interrupt. If this boolean is set to FALSE,
+  // the interrupt should be ignored.
+  BOOLEAN                        StackCookies;
+
+  // Indicates if the Memory Attribute Protocol (UEFI Specification 2.10) should be installed
+  //
+  // If TRUE, the protocol will be installed in CpuDxe. If FALSE, it will not be installed.
+  BOOLEAN                        InstallMemoryAttributeProtocol;
+} DXE_MEMORY_PROTECTION_SETTINGS;
+
+#define HOB_DXE_MEMORY_PROTECTION_SETTINGS_GUID \
+  { \
+    { 0x9ABFD639, 0xD1D0, 0x4EFF, { 0xBD, 0xB6, 0x7E, 0xC4, 0x19, 0x0D, 0x17, 0xD5 } } \
+  }

```

Comment:
nit: It's nice to have the GUID near the top of the file for files in `Include/Guid` so if someone is looking for the GUID they don't have to search far. Not a big deal though.

Reply:
Move to the top.